src/pydrex/minerals.py

```diff
@@ -76,109 +100,6 @@ def __iter__(self):
-    - ["Hirschmann2000"](https://doi.org/10.1029/2000GC000070)
-    - ["Herzberg2000"](https://doi.org/10.1029/2000GC000089)
-
-    """
-    match fit:
-        case "Herzberg2000":
-            return 1086 - 5.7 * pressure + 390 * np.log(pressure)
-        case "Hirschmann2000":
-            return 5.104 * pressure**2 + 132.899 * pressure + 1120.661
-        case _:
-            raise ValueError("unsupported fit")
-
-
-# TODO: Compare to [Man & Huang, 2011](https://doi.org/10.1007/s10659-011-9312-y).
-def voigt_averages(minerals, phase_assemblage, phase_fractions):
-    """Calculate elastic tensors as the Voigt averages of a collection of `mineral`s.
-
-    - `minerals` — list of `pydrex.minerals.Mineral` instances storing orientations and
-      fractional volumes of the grains within each distinct mineral phase
-    - `phase_assemblage` — collection of `pydrex.core.MineralPhase`s
-    - `phase_fractions` — collection of volume fractions for each phase in
-      `phase_assemblage` (values should sum to 1).
-
-    Raises a ValueError if the minerals contain an unequal number of grains or stored
-    texture results.
-
-    """
-    n_grains = minerals[0].n_grains
-    if not np.all([m.n_grains == n_grains for m in minerals[1:]]):
-        raise ValueError("cannot average minerals with unequal grain counts")
-    n_steps = len(minerals[0].orientations)
-    if not np.all([len(m.orientations) == n_steps for m in minerals[1:]]):
-        raise ValueError(
-            "cannot average minerals with variable-length orientation arrays"
-        )
-    if not np.all([len(m.fractions) == n_steps for m in minerals]):
-        raise ValueError(
-            "cannot average minerals with variable-length grain volume arrays"
-        )
-
-    elastic_tensors = StiffnessTensors()
-
-    # TODO: Perform rotation directly on the 6x6 matrices, see Carcione 2007.
-    # This trick is implemented in cpo_elastic_tensor.cc in Aspect.
-    average_tensors = np.zeros((n_steps, 6, 6))
-    for i in range(n_steps):
-        for mineral in minerals:
-            for n in range(n_grains):
-                match mineral.phase:
-                    case _core.MineralPhase.olivine:
-                        average_tensors[i] += _tensors.elastic_tensor_to_voigt(
-                            _tensors.rotate(
-                                elastic_tensors.olivine,
-                                mineral.orientations[i][n, ...].transpose(),
-                            )
-                            * mineral.fractions[i][n]
-                            * phase_fractions[phase_assemblage.index(mineral.phase)]
-                        )
-                    case _core.MineralPhase.enstatite:
-                        average_tensors[i] += _tensors.elastic_tensor_to_voigt(
-                            _tensors.rotate(
-                                elastic_tensors["enstatite"],
-                                minerals.orientations[i][n, ...].transpose(),
```

Comment:
Old enstatite code, tries to use a key instead of member, and `minerals` instead of `mineral`.